freqtrade/data/history/history_utils.py

```diff
@@ -295,6 +299,7 @@ def _download_trades_history(exchange: Exchange,
         # Default since_ms to 30 days if nothing is given
         new_trades = exchange.get_historic_trades(pair=pair,
                                                   since=since,
+                                                  until=until,
```

Comment:
i would think we should also have a check to verify if the end is before the "current data start date" (it's in the non-modified section between the 2 modified lines.

I'm also not certain this fully fixes #4412 - this only implements it for data-downloading of trades data ("kraken mode") ... but won't stop ohlcv downloads (binance, ftx, ...).

Reply:
> i would think we should also have a check to verify if the end is before the "current data start date" (it's in the non-modified section between the 2 modified lines). Otherwise we risk having people with "gaps" in the data - which will be difficult to explain (and even more difficult to fix)

I think the existing behaviour to clear the existing data if the start date is before the current data start date will be sufficient to prevent this?

> I'm also not certain this fully fixes #4412 - this only implements it for data-downloading of trades data ("kraken mode") ... but won't stop ohlcv downloads (binance, ftx, ...).

This is true.  Fixing it for "kraken mode" was really my main goal, since can take many hours (or days), compared to the seconds or minutes required for the OHLC version.  Would be nice to have both cases covered of course, but this one seemed like an easy win.

Reply:
> I think the existing behaviour to clear the existing data if the start date is before the current data start date will be sufficient to prevent this?

you're right ... i was thinking about the following behaviour

```
download-data --timerange 20200101-20200301

download-data --timerange 20200401-20200801
download-data --timerange 202001001-20201201
```

which at first glance, will skip March and August+September.
After thinking about it though - the 2nd call will adjust the startdate to start on March 1st ... and the 3rd call will also be adjusted.